beefy-gadget/src/worker.rs

```diff
@@ -346,15 +420,69 @@ where
 		}
 	}
 
+	fn send_outgoing_dkg_messages(&mut self) {
+		debug!(target: "beefy", "🕸️ Try sending DKG messages");
+		let authority_id = if let Some(id) = self.key_store.authority_id(self.rounds.validators().as_slice()) {
+			debug!(target: "beefy", "🕸️  Local authority id: {:?}", id);
+			id
+		} else {
+			panic!("error");
+		};
+		if let Some(curr_dkg) = self.dkg_state.curr_dkg.as_mut() {
+			curr_dkg.proceed();
+
+			if let Some(outgoing_messages) = curr_dkg.get_outgoing_message(&authority_id) {
```

Comment:
Do we know how reliable the gossip_engine's broadcast mechanism is? Curious if we have a way to handle messages not being received or processed properly and mistakenly end up losing them after clearing.

Reply:
Not very sure but I think current upstream project's configuration of libp2p gossip is reliable to deliver messages to all online peer for a subscribed topic. We can add store and re-broadcast logic if some issue comes in QA.